pkg/operator/certrotation/target_test.go

```diff
@@ -190,6 +201,132 @@ func TestEnsureTargetCertKeyPair(t *testing.T) {
 				}
 
 				actual := actions[1].(clienttesting.UpdateAction).GetObject().(*corev1.Secret)
+				if len(actual.Annotations) == 0 {
+					t.Errorf("expected certificates to be annotated")
+				}
+				ownershipValue, found := actual.Annotations[annotations.OpenShiftComponent]
+				if !found {
+					t.Errorf("expected secret to have ownership annotations, got: %v", actual.Annotations)
+				}
+				if ownershipValue != "test" {
+					t.Errorf("expected ownership annotation to be 'test', got: %v", ownershipValue)
+				}
+				if len(actual.Data["tls.crt"]) == 0 || len(actual.Data["tls.key"]) == 0 {
+					t.Error(actual.Data)
+				}
+				if actual.Annotations[CertificateHostnames] != "bar,foo" {
+					t.Error(actual.Annotations[CertificateHostnames])
+				}
+
+			},
+		},
+		{
+			name: "update SecretTLSType secrets",
+			caFn: func() (*crypto.CA, error) {
+				return newTestCACertificate(pkix.Name{CommonName: "signer-tests"}, int64(1), metav1.Duration{Duration: time.Hour * 24 * 60}, time.Now)
+			},
+			initialSecretFn: func() *corev1.Secret {
+				caBundleSecret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "target-secret", ResourceVersion: "10"},
+					Data:       map[string][]byte{},
+					Type:       "SecretTypeTLS",
+				}
+				return caBundleSecret
+			},
+			verifyActions: func(t *testing.T, client *kubefake.Clientset) {
+				actions := client.Actions()
+				if len(actions) != 5 {
+					t.Fatal(spew.Sdump(actions))
+				}
+
+				if !actions[0].Matches("get", "secrets") {
+					t.Error(actions[0])
+				}
+				if !actions[1].Matches("delete", "secrets") {
+					t.Error(actions[1])
+				}
+				if !actions[2].Matches("create", "secrets") {
+					t.Error(actions[0])
```

Comment:
There are few more of these:

```suggestion
					t.Error(actions[2])
```